components/encoder/single_encoder.go

```diff
@@ -143,10 +164,15 @@ func (e *SingleEncoder) Start(ctx context.Context) {
 			}
 
 			if e.m != nil {
+				// There is a minor race condition here. Delays in interrupt processing may result in a
+				// DirectionMoving() value that is *currently* different from one that was used to drive
+				// the motor. This may result in ticks being lost or applied in the wrong direction.
 				dir := e.m.DirectionMoving()
 				if dir == 1 || dir == -1 {
 					atomic.AddInt64(&e.position, dir)
 				}
+			} else {
+				e.logger.Warn("%s: received tick for encoder that isn't connected to a motor, ignoring.", e.name)
```

Comment:
not sure if warn is appropriate level. this should never happen if configured properly. thoughts?

Reply:
Can we check for directional awareness earlier? GoFor thinking you have information on the motor's position If we don't have a motor or direction pin, this model can't give you useful information outside of velocity which we don't expose. 

In general, you can attach encoders to passive joints (not attached to a motor) on a robot. This is used a lot in parallel mechanisms. This particular model can't actually do anything there though, since it can't sense direction and position without a motor.

Reply:
Ugh, the check for directional awareness happens in `motor_encoder.go:75` I think. There's a bit of circular logic going on here.

Reply:
I don't think we have another better choice, so I think this is okay. This will show GoFor and GoTo in app but not update position?